lib/amazon_mws.rb

```diff
@@ -12,7 +12,7 @@
 
 class AmazonMwsOrderResponse
   def initialize(response)
-    @response = response.fetch("GetOrderReferenceDetailsResponse", {})
+    @response = Hash.from_xml(response.body).fetch("GetOrderReferenceDetailsResponse", {})
```

Comment:
GetOrderReferenceDetails model was never created though, I just didn't think it was worth it since there would have to be a lot of customization